addons/stock/static/src/fields/stock_move_line_x2_many_field.js

```diff
@@ -15,6 +17,16 @@ export class SMLX2ManyField extends X2ManyField {
             onCreateEdit: () => this.createOpenRecord(),
         });
 
+        onMounted(async () => {
+            const orm = this.env.model.orm;
+            const domain = [
+                ["product_id", "=", this.props.record.data.product_id[0]],
+                ["location_id", "child_of", this.props.context.default_location_id],
```

Comment:
This will load all quants, potentially a big number for serial products.
Maybe something like
```
import { Domain } from "@web/core/domain";
...
            if(this.props.record.data.move_line_ids.records.length) {
                const domains = [];
                for(const ml of this.props.record.data.move_line_ids.records) {
                    domains.push([
                        ["product_id", "=", ml.data.product_id[0]],
                        ["lot_id", "=", ml.data.lot_id],
                        ["location_id", "=", ml.data.location_id[0]],
                        ["package_id", "=", ml.data.package_id],
                        ["owner_id", "=", ml.data.owner_id],
                    ]);
                }
                const quantIds = await orm.search("stock.quant", Domain.or(domains).toList());
                this.quantsData = await orm.read("stock.quant", quantIds, ['available_quantity']);
            }
            else {
                this.quantsData = null;
            }
```
to load only 'used' quants at open

@Whenrow Do we go on like this ?

Reply:
@naja628 @ajf-odoo It's ok to me to look for quant having some quantity only. 
Side note, you can group the 2 orm requests in one `orm.searchRead()`